tracer/build/_build/Build.Profiler.Steps.cs

```diff
@@ -38,9 +39,88 @@ partial class Build
         .DependsOn(CompileProfilerNativeTestsWindows)
         .DependsOn(CompileProfilerNativeTestsLinux);
 
+    Target SetupVcpkg => _ => _
+        .Unlisted()
+        .OnlyWhenStatic(() => IsWin)
+        .Executes(async () =>
+        {
+            var vcpkg = await GetVcpkg();
+
+            var vcpkgObjFolder = NativeBuildDirectory / "vcpkg";
+            var buildtreesRoot = vcpkgObjFolder / "buildtrees";
+            var packagesRoot = vcpkgObjFolder / "packages";
+            var downloadRoot = vcpkgObjFolder / "download";
+            vcpkg($"integrate install --x-buildtrees-root={buildtreesRoot} --x-packages-root={packagesRoot} --downloads-root={downloadRoot}");
+
+            async Task<Tool> GetVcpkg()
+            {
+                var vcpkgFilePath = string.Empty;
+
+                try
+                {
+                    vcpkgFilePath = ToolPathResolver.GetPathExecutable("vcpkg.exe");
+                }
+                catch (ArgumentException)
+                { }
+
+                if (File.Exists(vcpkgFilePath))
+                {
+                    return ToolResolver.GetLocalTool(vcpkgFilePath);
+                }
+
+                // Check if already downloaded
+                var vcpkgRoot = RootDirectory / "vcpkg";
+                var vcpkgExecPath = vcpkgRoot / "vcpkg.exe";
+
+                if (File.Exists(vcpkgExecPath))
+                {
+                    return ToolResolver.GetLocalTool($"{vcpkgExecPath}");
+                }
+
+                await DownloadAndExtractVcpkg(vcpkgRoot);
+                Cmd.Value(arguments: $"cmd /c {vcpkgRoot / "bootstrap-vcpkg.bat"}");
+                return ToolResolver.GetLocalTool($"{vcpkgRoot / "vcpkg.exe"}");
+            }
+
+            async Task DownloadAndExtractVcpkg(string destinationFolder)
+            {
+                var nbTries = 0;
+                var keepTrying = true;
+                var vcpkgZip = TempDirectory / "vcpkg.zip";
+                using var client = new HttpClient();
+                const string vcpkgVersion = "2024.11.16";
```

Comment:
Do we pin this version anywhere else? On the linux side for example? If not, I think this is fine, just wanted to check 🙂

Reply:
No, it's the only place where we fix.